content/academy/puppeteer_playwright/browser_contexts.md

```diff
@@ -0,0 +1,134 @@
+---
+title: VI - Creating multiple browser contexts
+description: Learn what a browser context is, how to create one, how to emulate devices, and how to use browser contexts to automate multiple sessions at one time.
+menuWeight: 7.6
+paths:
+    - puppeteer-playwright/browser-contexts
+---
+
+# [](#creating-browser-contexts) Creating multiple browser contexts
+
+A [**BrowserContext**](https://playwright.dev/docs/api/class-browsercontext) is an isolated incognito session within a **Browser** instance. This means that contexts can have different device/screen size configurations, different language and color scheme settings, etc. It is useful to use multiple browser instances when dealing with automating logging into multiple accounts simultaneously (therefore requiring multiple sessions), or in any cases where multiple sessions are required.
+
+When we create a **Browser** object by using the `launch()` function, a single [browser context](https://playwright.dev/docs/browser-contexts) is automatically created (only in Puppeteer). In order to create more, we use the [`browser.newContext()`](https://playwright.dev/docs/api/class-browser#browser-new-context) function in Playwright, and [`browser.createIncognitoBrowserContext`](https://pptr.dev/#?product=Puppeteer&version=v14.1.0&show=api-browsercreateincognitobrowsercontextoptions) in Puppeteer.
```

Comment:
I think it would be great to explain the difference between a persistent and non-persistent browser contexts, how they treat cookies, local storage and caches and that Puppeteer creates a persistent context by default while Playwright creates a non-persistent one by default and there's a special launch function to create the persistent one.

Reply:
Doing a bit of research. Now I understand there are differences, but what are they?